server/build/webpack.js

```diff
@@ -75,8 +75,17 @@ export default async function createCompiler (dir, { hotReload = false, dev = fa
     )
   }
 
-  const babelRuntimePath = require.resolve('babel-runtime/package')
-  .replace(/[\\/]package\.json$/, '')
+  const resolveModule = (moduleName) => {
+    // Check if there's a local version of the module
+    try {
+      const path = `next/node_modules/${moduleName}`
```

Comment:
Yeah, I agree, but this is in essence what `next` was previously doing, just in a different way.

For example, if your files are in `project/pages/` and you require `react`, running `require.resolve(...)` from that file will always refer to `project/node_modules/react` by virtue of how node resolves paths. We need to rewrite these somehow.

Previously this was done by running `require.resolve('react')` from the `next/server/build/webpack.js` file, which resolves absolutely to `/...longpath../project/node_modules/next/node_modules/react`. What I'm doing here is essentially exactly the same thing, but instead of using extracting the last bit with something like a regex to get a relative path, I'm just doing it in a cleaner way.

Hopefully that makes sense... :)